frontend/src/app/applications-upload/applications-upload.page.ts

```diff
@@ -0,0 +1,193 @@
+          }
+        }
+      });
+  }
+  /**
+   * In this method timeout of 1 second is speified to have a better readability of the application upload.
+   */
+  sleep(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+  /**
+   * In this method a post request is made to the server.
+   */
+  sendPostRequest(inp: FormsData): Promise<void> {
+    return this.sleep(1000).then(() => {
+      const keyCompetence: Array<{
+        category: string;
+        name: string;
+        rating: number;
+      }> = [];
+      const education: Array<EducationInfoEntry> = [];
+      const images: Array<{ content: string; type: string }> = [];
+      /**
+       * key competencies are mapped according json format required by server.
+       */
+      Object.entries(inp.keyCompetencies).map(([k, v]) => {
+        v.forEach((x: KeyCompetenciesEntry) => {
+          keyCompetence.push({ category: k, name: x.name, rating: x.rating });
+        });
+      });
+      /**
+       * education information are mapped according json format required by server.
+       */
+      Object.values(inp.educationInfo.educationInfoForm).map((v) => {
+        education.push(v);
+      });
+      /**
+       * picture of applicant is mapped according json format required by server.
+       */
+      images.push({
+        content: inp.profilePicture.pictureBase64,
+        type: 'profilePic',
+      });
+      /**
+       * complete json structure is formed here to send to server.
+       */
+      const formsData = {
+        firstName: inp.basicInfo.firstName,
+        lastName: inp.basicInfo.lastName,
+        phone: inp.contactInfo.phoneNumber,
+        email: inp.contactInfo.eMail,
+        title: inp.basicInfo.salutation,
+        linkedIn: inp.contactInfo.linkedIn,
+        xing: inp.contactInfo.xing,
+        imageList: images,
+        educations: education,
+        industries: inp.fieldDesignationInfo.field,
+        positions: inp.fieldDesignationInfo.designation,
+        keyCompetencies: keyCompetence,
+      };
+
+      const headers = new HttpHeaders();
+      headers.append('Content-Type', 'application/json');
+      /**
+       * post request is made here and based on response locally stored data is deleted
+       * error and success messages are shown wherever necessary
+       */
+      this.httpClient
+        .post(this.url + '/api/controller/createApplicant', formsData, {
+          responseType: 'text',
+          observe: 'response',
+        })
+        .subscribe(
+          (response) => {
+            if (response.status === 201) {
+              this.storage.deleteItem(this.storage.applicantDetailsDb, inp.id);
+              this.storage.deleteItem(this.storage.applicantRatingsDb, inp.id);
+              if (this.uploadSize === this.totalSize) {
+                this.isSuccess = true;
+                this.completionAlert();
+              } else {
+                this.uploadSize++;
```

Comment:
In html text it says uploading so shouldn't be incremented after sending API? Now it corresponds to uploaded count.

Reply:
It actually doing the same. Only on success of previous API, it is being incremented.